Cabal/src/Distribution/Simple/Setup/Build.hs

```diff
@@ -125,6 +127,13 @@ buildOptions progDb showOrParseArgs =
   [ optionNumJobs
       buildNumJobs
       (\v flags -> flags{buildNumJobs = v})
+  , option
+      []
+      ["semaphore"]
+      "semaphore"
```

Comment:
Could we have a `cabal build --help` to see how this is documented?

Reply:
```
 --semaphore                    Use a semaphore so GHC can compile components
                                in parallel
 -j, --jobs[=NUM]               Run NUM jobs simultaneously (or '$ncpus' if no
                                NUM is given).
```